.github/workflows/release.yml

```diff
@@ -1,30 +1,207 @@
+      - name: Set up the environment
+        uses: ./.github/actions/setup-env
 
       - name: Install Dependencies
         run: pnpm install --frozen-lockfile --prefer-offline
 
       - name: Create Release Pull Request
+        id: pr
+        env:
+          # NOTE: If we use the GITHUB_TOKEN to create the release PR, the checks will not be triggered automatically
+          GITHUB_TOKEN: ${{ github.token }}
         uses: changesets/action@v1
+        with:
+          version: pnpm run version
+
+      - name: Build All Packages
+        if: steps.pr.outputs.hasChangesets == 'false'
+        run: pnpm run --recursive -no-bail --filter './v-next/**' --if-present build
+
+      - name: Publish All Packages (dry-run)
+        if: steps.pr.outputs.hasChangesets == 'false'
+        run: pnpm publish --filter "./v-next/**" -r --no-git-checks --tag next --access public --dry-run
+
+      # NOTE: When running publish on an already published package we expect it to succeed but not to release anything and not include + hardhat@ in the output.
+      # If it does, however, include + hardhat@ in the output, it's OK, because we'll just update the release without moving the release tag forward.
+      - name: Publish All Packages
+        id: publish
+        if: steps.pr.outputs.hasChangesets == 'false'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NODE_AUTH_TOKEN }}
+          NPM_CONFIG_PROVENANCE: true
+        run: |
+          echo "stdout<<EOF" >> $GITHUB_OUTPUT
+          pnpm publish --filter "./v-next/**" -r --no-git-checks --tag next --access public | tee -a $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+      - name: Check if hardhat was published
+        id: hardhat
+        if: steps.pr.outputs.hasChangesets == 'false'
+        env:
+          STDOUT: ${{ steps.publish.outputs.stdout }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const lines = process.env.STDOUT.split('\n')
+
+            core.info('Checking if Hardhat was published')
+
+            const line = lines.find(line => line.startsWith('+ hardhat@'))
+
+            if (line === undefined) {
+              core.info('Hardhat was not published')
+              core.setOutput('version', '')
+              process.exit(0)
+            }
+
+            const version = line.split('@')[1]
+            core.info(`Hardhat was published with version ${version}`)
+            core.setOutput('version', version)
+
+      - name: Check the version of the published package
+        id: version
+        if: steps.hardhat.outputs.version != ''
+        env:
+          VERSION: ${{ steps.hardhat.outputs.version }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const version = process.env.VERSION
+
+            core.info('Checking if Hardhat was published as a prerelease')
+
+            // NOTE: This check would mark non-prerelease versions with - in the build tag as prereleases
```

Comment:
We don't use build tags in our versions - the `+BUILD_TAG` suffixes - so I think this should be enough for our usecase.